types/base.d.ts

```diff
@@ -9,11 +7,10 @@ export type ErrorValidator
 export interface Observable {
 	subscribe(observer: (value: {}) => void): void;
 }
-
 export type Test = (t: TestContext) => PromiseLike<void> | Iterator<any> | Observable | void;
-export type ContextualTest = (t: ContextualTestContext) => PromiseLike<void> | Iterator<any> | Observable | void;
+export type GenericTest<T> = (t: GenericTestContext<T>) => PromiseLike<void> | Iterator<any> | Observable | void;
```

Comment:
Add `export type ContextualTest = GenericTest<{ context: any }>;`, otherwise this might break some users. Same for `ContextualCallbackTest`